CSharp/Microsoft_Teams_Graph_RESTAPIs_Connect/Microsoft_Teams_Graph_RESTAPIs_Connect/ImportantFiles/GraphService.cs

```diff
@@ -115,9 +128,92 @@ public class GraphService
 
             return response;//response.ReasonPhrase;
         }
+
+        public async Task<string> CreateNewGroup(string accessToken, Group group)
+        {
+            string endpoint = ServiceHelper.GraphRootUri + "groups";
+            if (group != null)
+            {
+                group.groupTypes = new string[] { "Unified" };
+                group.mailEnabled = true;
+                group.securityEnabled = false;
+                group.visibility = "Private";
+            }
+
+
+            HttpResponseMessage response = await ServiceHelper.SendRequest(HttpMethod.Post, endpoint, accessToken, group);
+            if (response != null && response.IsSuccessStatusCode)
+            {
+                return Resource.TeamsGraph_CreateChannel_Success_Result;
+            }
+
+            await CreateTeam("", accessToken);
+            return response.ReasonPhrase;
+        }
+
+        public async Task<String> CreateTeam(string teamId, string accessToken)
+        {
+            string endpoint = ServiceHelper.GraphRootUri + "groups/" + teamId + "/team";
+            Team team = new Models.Team();
+            team.TeamGuestSettings = new Models.TeamGuestSettings() { allowCreateUpdateChannels = false, allowDeleteChannels = false };
+
+            HttpResponseMessage response = await ServiceHelper.SendRequest(HttpMethod.Post, endpoint, accessToken, team);
+            if (!response.IsSuccessStatusCode)
+                throw new Exception(response.ReasonPhrase);
+            return response.ReasonPhrase;
+        }
+
+
+        public async Task<String> UpdateTeam(string teamId, string accessToken)
+        {
+            string endpoint = ServiceHelper.GraphRootUri + "groups/" + teamId + "/team";
+            Team team = new Models.Team();
+            team.TeamGuestSettings = new Models.TeamGuestSettings() { allowCreateUpdateChannels = false, allowDeleteChannels = false };
+
+            team.memberSettings = new Models.TeamMemberSettings() { allowCreateUpdateChannels = false, allowDeleteChannels = false,allowAddRemoveApps=true, allowCreateUpdateRemoveConnectors=false, allowCreateUpdateRemoveTabs=true };
+            team.messagingSettings = new Models.TeamMessagingSettings() { allowChannelMentions = false, allowOwnerDeleteMessages = false,allowTeamMentions=true,allowUserDeleteMessages=true, allowUserEditMessages=false };
+            team.funSettings = new Models.TeamFunSettings() { giphyContentRating = "strict", allowGiphy = true };
+
+            HttpResponseMessage response = await ServiceHelper.SendRequest(new HttpMethod("Patch"), endpoint, accessToken, team);
+            if (!response.IsSuccessStatusCode)
+                throw new Exception(response.ReasonPhrase);
+            return response.ReasonPhrase;
+        }
+
+
+
+
+        public async Task AddMember(string teamId, Member member, string accessToken)
```

Comment:
There's a couple problems here. First, despite what I said earlier it's not actually necessary to turn a upn into an id, you can pass a UPN directly to /groups/{teamId}/members/$ref.

Second, the "Member" class is problematic. There is no such resource type in graph – there's User, and it doesn't have an owner property. I suggest changing AddMember() to take a upn and bool isOwner property.

Reply:
Also, for consistency we should call it groupId not teamId